Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -388,24 +408,42 @@ void GameListCtrl::BrowseForDirectory()
     {
       SConfig::GetInstance().m_ISOFolder.push_back(sPath);
       SConfig::GetInstance().SaveSettings();
-      ReloadList();
+      m_scan_trigger.Set();
     }
   }
 }
 
-void GameListCtrl::ReloadList()
+void GameListCtrl::RefreshList()
 {
   int scrollPos = wxWindow::GetScrollPos(wxVERTICAL);
   // Don't let the user refresh it while a game is running
   if (Core::GetState() != Core::State::Uninitialized)
     return;
 
-  ScanForISOs();
+  m_shown_files.clear();
+  for (auto& item : m_cached_files)
+  {
+    if (ShouldDisplayGameListItem(*item))
+      m_shown_files.push_back(item);
+  }
+
+  // Drives are not cached. Not sure if this is required, but better to err on the
+  // side of caution if cross-platform issues could come into play.
+  if (SConfig::GetInstance().m_ListDrives)
+  {
+    const Core::TitleDatabase title_database;
```

Comment:
That doesn't seem like a particularly important/common use case.  
The TitleDatabase instance could at least be made a member variable of GameListCtrl, and only reparse the files on a full rescan event.